caddyhttp/basicauth/basicauth.go

```diff
@@ -106,10 +107,10 @@ func (a BasicAuth) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error
 
 		// Get a replacer so we can provide basic info for the authentication error.
 		repl := httpserver.NewReplacer(r, nil, "-")
-		errstr := repl.Replace("BasicAuth: user \"%s\" was not found or password was incorrect. {remote} {host} {uri} {proto}")
+		repl.Set("user", username)
+		errstr := repl.Replace("BasicAuth: user \"{user}\" was not found or password was incorrect. {remote} {host} {uri} {proto}")
 
-		// Username will not exist in Replacer so provide here.
-		err := fmt.Errorf(errstr, username)
+		err := errors.New(errstr)
```

Comment:
Would prefer fmt.Errorf actually, for consistency with most of the rest of Caddy. :)

Reply:
Is it ok to call Errorf even if there are no vars in the format string?  Are you happy with the rest of the change?